src/sotauptaneclient.cc

```diff
@@ -21,8 +21,8 @@ void SotaUptaneClient::run(command::Channel *commands_channel) {
 }
 
 bool SotaUptaneClient::isInstalled(const Uptane::Target &target) {
-  if (target.ecu_identifier() == config.uptane.primary_ecu_serial) {
-    return target.filename() == OstreePackage::getEcu(config.uptane.primary_ecu_serial, config.ostree.sysroot).ref_name;
+  if (target.ecu_identifier() == uptane_repo.primary_ecu_serial) {
+    return target.filename() == OstreePackage::getEcu(uptane_repo.primary_ecu_serial, config.ostree.sysroot).ref_name;
```

Comment:
primary_ecu_serial is private in Uptane::Repository, and we can only do this because SotaUpdateClient is a friend of Uptane::Respoitory.  Can we stop them getting too entwined?  We are quite close now to being able to stop them being 'friends', so perhaps this change belongs in a separate commit, where we can figure out how to signing can work